crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -425,3 +428,20 @@ pub struct UnsupportedLocation {
     #[help]
     pub help: Option<String>,
 }
+
+#[derive(Diagnostic, Debug, Error, Clone, Hash, PartialEq, Eq)]
+#[error("Subscription operations can not have an introspection field as a root field")]
+#[diagnostic(code("apollo-compiler validation error"))]
+pub struct IntrospectionField {
+    // current definition
+    pub field: String,
+
+    #[help]
+    pub help: Option<String>,
```

Comment:
i think `help` field is not necessary here. The `error` above describes the problem, and the label for definition describes what's wrong with this field.